docfiles/tracking.html

```diff
@@ -2,54 +2,68 @@
+                overridePageViewDuration: false,
+                disableExceptionTracking: true,
+                disableCookiesUsage: !isProduction,
+                isStorageUseDisabled: !isProduction,
+                url: "/doccdn/ai.2.min.js"
+            }
         });
         window.appInsights=appInsights;
-        appInsights.queue.push(function () {
-            appInsights.context.addTelemetryInitializer(function (envelope) {
-                var telemetryItem = envelope.data.baseData;
-                telemetryItem.properties = telemetryItem.properties || {};
-                telemetryItem.properties["cookie"] = isProduction;
-                if (typeof pxtConfig === "undefined" || !pxtConfig || !pxtConfig.targetId) {
-                    telemetryItem.properties["target"] = "@targetid@";
-                    return;
-                }
-                telemetryItem.properties["target"] = pxtConfig.targetId;
-                telemetryItem.properties["stage"] = (pxtConfig.relprefix || "/--").replace(/[^a-z]/ig, '')
-                if (typeof Windows !== "undefined")
-                    telemetryItem.properties["WindowsApp"] = 1;
-                var userAgent = navigator.userAgent.toLowerCase();
-                var userAgentRegexResult = /\belectron\/(\d+\.\d+\.\d+.*?)(?: |$)/i.exec(userAgent); // Example navigator.userAgent: "Mozilla/5.0 Chrome/61.0.3163.100 Electron/2.0.0 Safari/537.36"
-                if (userAgentRegexResult) {
-                    telemetryItem.properties["Electron"] = 1;
-                    telemetryItem.properties["ElectronVersion"] = userAgentRegexResult[1];
-                }
-                if (typeof pxtElectron !== "undefined") {
-                    telemetryItem.properties["PxtElectron"] = 1;
-                    telemetryItem.properties["ElectronVersion"] = pxtElectron.versions.electronVersion;
-                    telemetryItem.properties["ChromiumVersion"] = pxtElectron.versions.chromiumVersion;
-                    telemetryItem.properties["NodeVersion"] = pxtElectron.versions.nodeVersion;
-                    telemetryItem.properties["PxtElectronVersion"] = pxtElectron.versions.pxtElectronVersion;
-                    telemetryItem.properties["PxtCoreVersion"] = pxtElectron.versions.pxtCoreVersion;
-                    telemetryItem.properties["PxtTargetVersion"] = pxtElectron.versions.pxtTargetVersion;
-                    telemetryItem.properties["PxtElectronIsProd"] = pxtElectron.versions.isProd;
-                }
-            });
-        });
         var location = window.location.toString();
         // Use generic title for script share page, otherwise use current page title
         var pageTitle = isScriptPage(location) ? "Share Page" : null;
-        appInsights.trackPageView(pageTitle, scrubUrl(location), {urlReferrer: scrubUrl(document.referrer.toString())});
+        appInsights.trackPageView({
+            name: pageTitle,
+            uri: scrubUrl(location),
```

Comment:
Can we send existing name and uri as part of properties and set the top level name and uri to something like following? 

pageTitle: "Share Page"
uri: "arcade.makecode.com"  (Just the domain name).

This is to avoid explosion of properties to help data team. I can explain

Reply:
Updated it to use window.location.hostname for the pageTitle and window.location.origin for the uri, which leads to names like "makecode.microbit.org" and urls like "https://makecode.microbit.org". It's a little redundant but I'm hopeful it'll at least stop the property "explosion" issue. Does that seem sufficient?

![image](https://user-images.githubusercontent.com/69657545/234723664-b382825b-a612-47e0-a7bb-aa6b1081018d.png)

Microbit target with this change: https://makecode.microbit.org/app/9bc086c4340afea1f95376a4d20278faac5e4839-cdf103f54d
Arcade target with this change: https://arcade.makecode.com/app/bd62aa9e2785e9a3effbb7957c7ad33cb29ae27c-74d4bebe50

Reply:
Also visible in that screenshot: original url moved into properties